pandaserver/dataservice/adder_gen.py

```diff
@@ -117,411 +117,461 @@ def get_plugin_class(self, tmp_vo, tmp_group):
-                # use failed for cancelled/closed jobs
-                if self.job.isCancelled():
-                    self.job_status = "failed"
-                    # reset error codes to skip retrial module
-                    self.job.pilotErrorCode = 0
-                    self.job.exeErrorCode = 0
-                    self.job.ddmErrorCode = 0
-                # keep old status
-                old_job_status = self.job.jobStatus
-                # set job status
-                if self.job.jobStatus not in ["transferring"]:
-                    self.job.jobStatus = self.job_status
-                add_result = None
-                adder_plugin = None
-                # parse JSON
-                parse_result = self.parse_json()
-                if parse_result < 2:
-                    # interaction with DDM
+                self.logger.debug("apply_retrial_rules is back")
+            except Exception as e:
+                self.logger.error(
+                    f"apply_retrial_rules excepted and needs to be investigated ({e}): {traceback.format_exc()}")
+
+        self.job.jobStatus = "failed"
+        for file in self.job.Files:
+            if file.type in ["output", "log"]:
+                if self.add_result is not None and file.lfn in self.add_result.merging_files:
+                    file.status = "merging"
+                else:
+                    file.status = "failed"
+
+    def check_job_status(self) -> None:
+        """
+        Check the job status and log appropriate messages.
+        """
+        if self.job is None:
+            self.logger.debug(": job not found in DB")
+        elif self.job.jobStatus in ["finished", "failed", "unknown", "merging"]:
+            self.logger.error(f": invalid state -> {self.job.jobStatus}")
+        elif self.attempt_nr is not None and self.job.attemptNr != self.attempt_nr:
+            self.logger.error(f"wrong attemptNr -> job={self.job.attemptNr} <> {self.attempt_nr}")
+        elif self.job_status == EventServiceUtils.esRegStatus:
+            self.register_event_service_files()
+        else:
+            # check file status in JEDI
+            self.check_file_status_in_jedi()
+
+            # use failed for cancelled/closed jobs
+            if self.job.isCancelled():
+                self.job_status = "failed"
+                # reset error codes to skip retrial module
+                self.job.pilotErrorCode = 0
+                self.job.exeErrorCode = 0
+                self.job.ddmErrorCode = 0
+
+            # keep old status
+            old_job_status = self.job.jobStatus
+
+            # set job status
+            if self.job.jobStatus not in ["transferring"]:
+                self.job.jobStatus = self.job_status
+            self.add_result = None
+            adder_plugin = None
+
+            # parse JSON
+            parse_result = self.parse_json()
+
+            if parse_result < 2:
```

Comment:
what is < 2?

Reply:
If the parse_result is less than 2, it means that the parsing either succeeded or encountered a harmless error. In these cases, the function proceeds to execute the plugin by calling self.execute_plugin().
It is explained in the method:

`        :return: An integer indicating the result of the parsing process.
                 0 - succeeded
                 1 - harmless error to exit
                 2 - fatal error
                 3 - event service
`
I will add a comment

Reply:
Yes, I know, but it would be easier to read if you put a comment here. Or, you make constants
OK = 0
HARMLESS_ERROR = 1
FATAL_ERROR = 2
EVENT_SERVICE = 3

if parse_result in [OK, HARMLESS_ERROR]:
 ...
 
 (and the returns in parse_json are return OK/HARMLESS_ERROR/FATAL_ERROR/EVENT_SERVICE...)